### DISCUSSION TITLE
Setting up DAP for Go (Delve)

### DISCUSSION THREAD

**Discussion by Jomik** [LazyVim/LazyVim#607] [Q&A] [answered]
What is the simplest way to setup `delve` ? I expected being able to just pass `delve` to `ensure_installed` and it would install and run the `automatic_setup` as well?
This does not seem to work for my case though. I actually do not even seem to get `delve` installed.
I added
```lua
  {
    "mfussenegger/nvim-dap",
    dependencies = {
      {
        "jay-babu/mason-nvim-dap.nvim",
        opts = {
          ensure_installed = {
            "delve",
          },
        },
      },
    },
  },
```

Do you want a `lazyvim.plugins.extras.dap.delve` (or `go`)?

**Comment by folke** [comment]
Just do:

```lua
return {
  {
    "mason-nvim-dap.nvim",
    opts = {
      ensure_installed = { "delve" },
    },
  },
}
```

**Reply by vadlungu** [reply]
Hello folks, 

how can I attach it to dap ui , so I could debug using terminal input from stdin.
someting like in vscode , and to have in dap-ui separate window like terminal where I could insert expected input

<details><summary> example </summary>

```json
{
    // See https://go.microsoft.com/fwlink/?LinkId=733558
    // for the documentation about the tasks.json format
    "version": "2.0.0",
    "tasks": [
        {
            "label": "delve",
            "type": "shell",
            "command": "dlv debug --headless --listen=:23456 --api-version=2 \"${workspaceFolder}\"",
            "isBackground": true,
            "presentation": {
                "focus": true,
                "panel": "dedicated",
                "clear": false
            },
            "group": {
                "kind": "build",
                "isDefault": true
            },
            "problemMatcher": {
                "pattern": {
                    "regexp": ""
                },
                "background": {
                    "activeOnStart": true,
                    "beginsPattern": {
                        "regexp": ".*"
                    },
                    "endsPattern": {
                        "regexp": ".*server listening.*"
                    }
                }
            }
        }
    ]
}
```

</details>

**Comment by folke** [comment, marked as answer]
Or do `:DapInstall delve`

**Comment by Jomik** [comment]
Weird. `:DapInstall delve` simply works. I would like to have it explicitly in my config though.
Adding `delve` to `opts.ensure_installed` for `mason-nvim-dap` does not work.
Adding it for just `mason` works though. Would like to have it under a `dap` plugin though 🤔 

Any idea what is causing this?

**Reply by folke** [reply]
That's odd. I added `chrome` and `python` and those got automatically installed for me

**Reply by folke** [reply]
You can find the mason-dap mappings here https://github.com/jay-babu/mason-nvim-dap.nvim/blob/main/lua/mason-nvim-dap/mappings/source.lua

Delve is in there though, so no idea why that wouldn't work

**Reply by folke** [reply]
Ah wait, it will only auto install when loaded, so when you set a breakpoint for example.
The whole dap config is lazy loaded.

**Reply by Jomik** [reply]
Shouldn't it load if I run `:DapInstall` ? It at least shows up as loaded in `:Lazy`.
